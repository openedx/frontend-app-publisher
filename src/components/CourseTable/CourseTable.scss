.width-percent-44{
    width: 44%;
}
.width-percent-12{
    width: 12%;
}
