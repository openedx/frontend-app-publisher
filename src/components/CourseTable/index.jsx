import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';
import React from 'react';
import { Link } from 'react-router-dom';
import Select from 'react-select';
import qs from 'query-string';
import { getAuthenticatedUser } from '@edx/frontend-platform/auth';
import { SearchField } from '@edx/paragon';

import TableContainer from '../../containers/TableContainer';
import ButtonToolbar from '../ButtonToolbar';
import PageContainer from '../PageContainer';
import { formatDate, getPageOptionsFromUrl, updateUrl } from '../../utils';
import Pill from '../Pill';
import { PUBLISHED, REVIEWED } from '../../data/constants';

const dot = color => ({
  alignItems: 'center',
  display: 'flex',

  ':before': {
    backgroundColor: color,
    borderRadius: 10,
    content: '" "',
    display: 'block',
    marginRight: 8,
    height: 10,
    width: 10,
  },
});

class CourseTable extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      filterGroups: [
        {
          label: 'Course Run Statuses',
          options: [
            { value: 'in_review', label: 'In review', color: '#e7e7e7' },
            { value: PUBLISHED, label: 'Published', color: '#008100' },
            { value: REVIEWED, label: 'Scheduled', color: '#0075b4' },
            { value: 'unsubmitted', label: 'Unsubmitted', color: '#E2C018' },
          ],
        },
      ],
      selectedFilters: [],
    };
  }

  componentDidMount() {
    this.props.fetchOrganizations();
  }

  componentDidUpdate(prevProps) {
    const {
      table: {
        editorFilterOptions,
      },
    } = this.props;
    const { administrator } = getAuthenticatedUser();
    const prevEditorFilterOptions = prevProps.table.editorFilterOptions;

    if (editorFilterOptions !== prevEditorFilterOptions && !administrator) {
      this.state.filterGroups.push({
        label: 'Course Editors',
        options: editorFilterOptions.map(editor => ({ value: editor.id, label: editor.name })),
      });
      this.getSelectedFiltersFromUrl();
    }
    const {
      editors: prevEditors,
      course_run_statuses: prevCourseRunStatuses,
    } = qs.parse(prevProps.location.search);
    const {
      editors,
      course_run_statuses: courseRunStatuses,
    } = qs.parse(this.props.location.search);
    if ((editors !== prevEditors) || (courseRunStatuses !== prevCourseRunStatuses)) {
      this.getSelectedFiltersFromUrl();
    }
  }

  getSelectedFiltersFromUrl() {
    const pageOptions = getPageOptionsFromUrl();

    const courseRunStatusesFromQuery = pageOptions.course_run_statuses
      ? pageOptions.course_run_statuses.split(',') : null;

    const editorsFromQuery = pageOptions.editors ? pageOptions.editors.split(',') : null;
    const selectedEditors = editorsFromQuery ? this.state.filterGroups.find(group => (
      group.label === 'Course Editors'
    )).options.filter(option => editorsFromQuery.includes(option.value.toString())) : [];

    if (!courseRunStatusesFromQuery) {
      this.setState({
        selectedFilters: selectedEditors,
      });
    } else {
      this.setState(prevState => ({
        selectedFilters: prevState.filterGroups.find(group => (
          group.label === 'Course Run Statuses'
        ))
          .options
          .filter(option => courseRunStatusesFromQuery.includes(option.value))
          .concat(selectedEditors),
      }));
    }
  }

  updateFilterQueryParamsInUrl(selectedFilters) {
    const courseRunStatusParams = selectedFilters.filter(filter => !Number.isInteger(filter.value));
    const editorParams = selectedFilters.filter(filter => Number.isInteger(filter.value));
    const params = {
      course_run_statuses: courseRunStatusParams.length
        ? courseRunStatusParams.map(filter => filter.value).toString() : null,
      editors: editorParams.length ? editorParams.map(filter => filter.value).toString() : null,
    };
    updateUrl({ ...params, page: 1 });
  }

  renderTableHeader() {
    const { selectedFilters, filterGroups } = this.state;
    const pageOptions = getPageOptionsFromUrl();

    return (
      <>
        <div className="row">
          <div className="col-2 float-left">
            <ButtonToolbar className="mb-3" leftJustify>
              <Link to="/courses/new">
                <button type="button" className="btn btn-primary">New Course</button>
              </Link>
            </ButtonToolbar>
          </div>
          <div className="col-5 float-right">
            <Select
              closeMenuOnSelect={false}
              value={selectedFilters}
              options={filterGroups}
              onChange={filters => this.updateFilterQueryParamsInUrl(filters === null
                ? [] : filters)}
              isMulti
              maxMenuHeight="30vh"
              placeholder="Filters..."
              styles={
                {
                  option: (styles, { data }) => ({ ...styles, ...dot(data.color) }),
                  multiValue: (styles, { data }) => (
                    { ...styles, backgroundColor: data.color || '#e7e7e7', opacity: 0.7 }
                  ),
                  multiValueLabel: (styles, { data }) => (
                    {
                      ...styles,
                      color: data.label === 'Published' || data.label === 'Scheduled' ? '#ffffff' : '#000000',
                    }
                  ),
                }
              }
            />
          </div>
          <div className="col-5 float-right">
            <SearchField
              value={pageOptions.pubq}
              onClear={() => {
                updateUrl({ filter: null });
              }}
              onSubmit={(filter) => {
                updateUrl({ filter, page: 1 });
              }}
              placeholder="Search"
            />
          </div>
        </div>
      </>
    );
  }

  render() {
    const {
      table: {
        error,
        loading,
      },
    } = this.props;

    const courseTableColumns = [
      {
        label: 'Course Name',
        key: 'title',
        columnSortable: true,
      },
      {
        label: 'Course Number',
        key: 'number',
        columnSortable: true,
      },
      {
        label: 'States',
        key: 'course_run_statuses',
        columnSortable: false,
      },
      {
        label: 'Course Editors',
        key: 'course_editor_names',
        columnSortable: false,
      },
    ];
    const formatCourseData = courses => courses.map(course => ({
      ...course,
      title: (<Link to={`/courses/${course.uuid}`}>{course.title}</Link>),
      modified: formatDate(course.modified),
      number: course.key_for_reruns || course.key,
      course_run_statuses: (<Pill statuses={course.course_run_statuses} />),
      course_editor_names: course.editors ? course.editors.map(editor => editor.user.full_name).join(', ') : '',
    }));

    return (
      <PageContainer wide>
        <Helmet>
          <title>{`Publisher | ${process.env.SITE_NAME}`}</title>
        </Helmet>
        {!loading && !error && this.renderTableHeader()}
        <TableContainer
          className="courses"
          columns={courseTableColumns}
          formatData={formatCourseData}
          tableSortable
        />
      </PageContainer>
    );
  }
}

CourseTable.defaultProps = {
  fetchOrganizations: () => {},
  publisherUserInfo: {
    organizations: [],
  },
  table: {
    error: null,
    loading: false,
    editorFilterOptions: [],
  },
};

CourseTable.propTypes = {
  fetchOrganizations: PropTypes.func,
  publisherUserInfo: PropTypes.shape({
    organizations: PropTypes.array,
    error: PropTypes.arrayOf(PropTypes.string),
    isFetching: PropTypes.bool,
  }),
  location: PropTypes.shape({
    search: PropTypes.string,
  }).isRequired,
  table: PropTypes.shape({
    error: PropTypes.arrayOf(PropTypes.string),
    loading: PropTypes.bool,
    editorFilterOptions: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number,
      full_name: PropTypes.string,
    })),
  }),
};

export default CourseTable;
