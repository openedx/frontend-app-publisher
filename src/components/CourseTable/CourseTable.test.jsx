import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CourseTable from './index';

describe('CourseTable', () => {
  it('shows a table', () => {
    const component = shallow(<CourseTable />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('hides table and button when blacklisted', () => {
    const publisherUserInfo = { organizations: [{ key: 'fake1', name: 'fake_name1' }] };
    const component = shallow(<CourseTable publisherUserInfo={publisherUserInfo} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('displays table and button when not blacklisted', () => {
    const publisherUserInfo = { organizations: [{ key: 'fake2', name: 'fake_name2' }] };
    const component = shallow(<CourseTable publisherUserInfo={publisherUserInfo} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('displays table and button when user has no orgs', () => {
    const publisherUserInfo = { organizations: [] };
    const component = shallow(<CourseTable publisherUserInfo={publisherUserInfo} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
