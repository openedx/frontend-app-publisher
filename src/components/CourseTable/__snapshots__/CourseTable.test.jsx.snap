// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseTable displays table and button when not blacklisted 1`] = `
<PageContainer
  className=""
  wide={true}
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Publisher | undefined
    </title>
  </HelmetWrapper>
  <div
    className="row"
  >
    <div
      className="col-2 float-left"
    >
      <ButtonToolbar
        className="mb-3"
        leftJustify={true}
      >
        <Link
          to="/courses/new"
        >
          <button
            className="btn btn-primary"
            type="button"
          >
            New Course
          </button>
        </Link>
      </ButtonToolbar>
    </div>
    <div
      className="col-5 float-right"
    >
      <StateManager
        closeMenuOnSelect={false}
        defaultInputValue=""
        defaultMenuIsOpen={false}
        defaultValue={null}
        isMulti={true}
        maxMenuHeight="30vh"
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "Course Run Statuses",
              "options": Array [
                Object {
                  "color": "#e7e7e7",
                  "label": "In review",
                  "value": "in_review",
                },
                Object {
                  "color": "#008100",
                  "label": "Published",
                  "value": "published",
                },
                Object {
                  "color": "#0075b4",
                  "label": "Scheduled",
                  "value": "reviewed",
                },
                Object {
                  "color": "#E2C018",
                  "label": "Unsubmitted",
                  "value": "unsubmitted",
                },
              ],
            },
          ]
        }
        placeholder="Filters..."
        styles={
          Object {
            "multiValue": [Function],
            "multiValueLabel": [Function],
            "option": [Function],
          }
        }
        value={Array []}
      />
    </div>
    <div
      className="col-5 float-right"
    >
      <SearchField
        icons={
          Object {
            "clear": <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    352,
                    512,
                    Array [],
                    "f00d",
                    "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
                  ],
                  "iconName": "times",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />,
            "submit": <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f002",
                    "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                  ],
                  "iconName": "search",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />,
          }
        }
        onBlur={[Function]}
        onChange={[Function]}
        onClear={[Function]}
        onFocus={[Function]}
        onSubmit={[Function]}
        placeholder="Search"
        screenReaderText={
          Object {
            "clearButton": "clear search",
            "label": "search",
            "submitButton": "submit search",
          }
        }
        value=""
      />
    </div>
  </div>
  <withRouter(Connect(TableComponent))
    className="courses"
    columns={
      Array [
        Object {
          "columnSortable": true,
          "key": "title",
          "label": "Course Name",
        },
        Object {
          "columnSortable": true,
          "key": "number",
          "label": "Course Number",
        },
        Object {
          "columnSortable": false,
          "key": "course_run_statuses",
          "label": "States",
        },
        Object {
          "columnSortable": false,
          "key": "course_editor_names",
          "label": "Course Editors",
        },
      ]
    }
    formatData={[Function]}
    tableSortable={true}
  />
</PageContainer>
`;

exports[`CourseTable displays table and button when user has no orgs 1`] = `
<PageContainer
  className=""
  wide={true}
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Publisher | undefined
    </title>
  </HelmetWrapper>
  <div
    className="row"
  >
    <div
      className="col-2 float-left"
    >
      <ButtonToolbar
        className="mb-3"
        leftJustify={true}
      >
        <Link
          to="/courses/new"
        >
          <button
            className="btn btn-primary"
            type="button"
          >
            New Course
          </button>
        </Link>
      </ButtonToolbar>
    </div>
    <div
      className="col-5 float-right"
    >
      <StateManager
        closeMenuOnSelect={false}
        defaultInputValue=""
        defaultMenuIsOpen={false}
        defaultValue={null}
        isMulti={true}
        maxMenuHeight="30vh"
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "Course Run Statuses",
              "options": Array [
                Object {
                  "color": "#e7e7e7",
                  "label": "In review",
                  "value": "in_review",
                },
                Object {
                  "color": "#008100",
                  "label": "Published",
                  "value": "published",
                },
                Object {
                  "color": "#0075b4",
                  "label": "Scheduled",
                  "value": "reviewed",
                },
                Object {
                  "color": "#E2C018",
                  "label": "Unsubmitted",
                  "value": "unsubmitted",
                },
              ],
            },
          ]
        }
        placeholder="Filters..."
        styles={
          Object {
            "multiValue": [Function],
            "multiValueLabel": [Function],
            "option": [Function],
          }
        }
        value={Array []}
      />
    </div>
    <div
      className="col-5 float-right"
    >
      <SearchField
        icons={
          Object {
            "clear": <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    352,
                    512,
                    Array [],
                    "f00d",
                    "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
                  ],
                  "iconName": "times",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />,
            "submit": <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f002",
                    "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                  ],
                  "iconName": "search",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />,
          }
        }
        onBlur={[Function]}
        onChange={[Function]}
        onClear={[Function]}
        onFocus={[Function]}
        onSubmit={[Function]}
        placeholder="Search"
        screenReaderText={
          Object {
            "clearButton": "clear search",
            "label": "search",
            "submitButton": "submit search",
          }
        }
        value=""
      />
    </div>
  </div>
  <withRouter(Connect(TableComponent))
    className="courses"
    columns={
      Array [
        Object {
          "columnSortable": true,
          "key": "title",
          "label": "Course Name",
        },
        Object {
          "columnSortable": true,
          "key": "number",
          "label": "Course Number",
        },
        Object {
          "columnSortable": false,
          "key": "course_run_statuses",
          "label": "States",
        },
        Object {
          "columnSortable": false,
          "key": "course_editor_names",
          "label": "Course Editors",
        },
      ]
    }
    formatData={[Function]}
    tableSortable={true}
  />
</PageContainer>
`;

exports[`CourseTable hides table and button when blacklisted 1`] = `
<PageContainer
  className=""
  wide={true}
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Publisher | undefined
    </title>
  </HelmetWrapper>
  <div
    className="row"
  >
    <div
      className="col-2 float-left"
    >
      <ButtonToolbar
        className="mb-3"
        leftJustify={true}
      >
        <Link
          to="/courses/new"
        >
          <button
            className="btn btn-primary"
            type="button"
          >
            New Course
          </button>
        </Link>
      </ButtonToolbar>
    </div>
    <div
      className="col-5 float-right"
    >
      <StateManager
        closeMenuOnSelect={false}
        defaultInputValue=""
        defaultMenuIsOpen={false}
        defaultValue={null}
        isMulti={true}
        maxMenuHeight="30vh"
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "Course Run Statuses",
              "options": Array [
                Object {
                  "color": "#e7e7e7",
                  "label": "In review",
                  "value": "in_review",
                },
                Object {
                  "color": "#008100",
                  "label": "Published",
                  "value": "published",
                },
                Object {
                  "color": "#0075b4",
                  "label": "Scheduled",
                  "value": "reviewed",
                },
                Object {
                  "color": "#E2C018",
                  "label": "Unsubmitted",
                  "value": "unsubmitted",
                },
              ],
            },
          ]
        }
        placeholder="Filters..."
        styles={
          Object {
            "multiValue": [Function],
            "multiValueLabel": [Function],
            "option": [Function],
          }
        }
        value={Array []}
      />
    </div>
    <div
      className="col-5 float-right"
    >
      <SearchField
        icons={
          Object {
            "clear": <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    352,
                    512,
                    Array [],
                    "f00d",
                    "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
                  ],
                  "iconName": "times",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />,
            "submit": <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f002",
                    "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                  ],
                  "iconName": "search",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />,
          }
        }
        onBlur={[Function]}
        onChange={[Function]}
        onClear={[Function]}
        onFocus={[Function]}
        onSubmit={[Function]}
        placeholder="Search"
        screenReaderText={
          Object {
            "clearButton": "clear search",
            "label": "search",
            "submitButton": "submit search",
          }
        }
        value=""
      />
    </div>
  </div>
  <withRouter(Connect(TableComponent))
    className="courses"
    columns={
      Array [
        Object {
          "columnSortable": true,
          "key": "title",
          "label": "Course Name",
        },
        Object {
          "columnSortable": true,
          "key": "number",
          "label": "Course Number",
        },
        Object {
          "columnSortable": false,
          "key": "course_run_statuses",
          "label": "States",
        },
        Object {
          "columnSortable": false,
          "key": "course_editor_names",
          "label": "Course Editors",
        },
      ]
    }
    formatData={[Function]}
    tableSortable={true}
  />
</PageContainer>
`;

exports[`CourseTable shows a table 1`] = `
<PageContainer
  className=""
  wide={true}
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Publisher | undefined
    </title>
  </HelmetWrapper>
  <div
    className="row"
  >
    <div
      className="col-2 float-left"
    >
      <ButtonToolbar
        className="mb-3"
        leftJustify={true}
      >
        <Link
          to="/courses/new"
        >
          <button
            className="btn btn-primary"
            type="button"
          >
            New Course
          </button>
        </Link>
      </ButtonToolbar>
    </div>
    <div
      className="col-5 float-right"
    >
      <StateManager
        closeMenuOnSelect={false}
        defaultInputValue=""
        defaultMenuIsOpen={false}
        defaultValue={null}
        isMulti={true}
        maxMenuHeight="30vh"
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "Course Run Statuses",
              "options": Array [
                Object {
                  "color": "#e7e7e7",
                  "label": "In review",
                  "value": "in_review",
                },
                Object {
                  "color": "#008100",
                  "label": "Published",
                  "value": "published",
                },
                Object {
                  "color": "#0075b4",
                  "label": "Scheduled",
                  "value": "reviewed",
                },
                Object {
                  "color": "#E2C018",
                  "label": "Unsubmitted",
                  "value": "unsubmitted",
                },
              ],
            },
          ]
        }
        placeholder="Filters..."
        styles={
          Object {
            "multiValue": [Function],
            "multiValueLabel": [Function],
            "option": [Function],
          }
        }
        value={Array []}
      />
    </div>
    <div
      className="col-5 float-right"
    >
      <SearchField
        icons={
          Object {
            "clear": <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    352,
                    512,
                    Array [],
                    "f00d",
                    "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
                  ],
                  "iconName": "times",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />,
            "submit": <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f002",
                    "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                  ],
                  "iconName": "search",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />,
          }
        }
        onBlur={[Function]}
        onChange={[Function]}
        onClear={[Function]}
        onFocus={[Function]}
        onSubmit={[Function]}
        placeholder="Search"
        screenReaderText={
          Object {
            "clearButton": "clear search",
            "label": "search",
            "submitButton": "submit search",
          }
        }
        value=""
      />
    </div>
  </div>
  <withRouter(Connect(TableComponent))
    className="courses"
    columns={
      Array [
        Object {
          "columnSortable": true,
          "key": "title",
          "label": "Course Name",
        },
        Object {
          "columnSortable": true,
          "key": "number",
          "label": "Course Number",
        },
        Object {
          "columnSortable": false,
          "key": "course_run_statuses",
          "label": "States",
        },
        Object {
          "columnSortable": false,
          "key": "course_editor_names",
          "label": "Course Editors",
        },
      ]
    }
    formatData={[Function]}
    tableSortable={true}
  />
</PageContainer>
`;
